src/internal/proc.js

```diff
@@ -600,6 +601,21 @@ export default function proc(
     cb(!!mainTask.isCancelled)
   }
 
+  function runFlushEffect(channel, cb) {
+    const bufferedItems = []
+    for (;;) {
+      let exposedItem
+      channel.take(item => exposedItem = item)
```

Comment:
Ye, you are right ofc. ;) I just remember now that's the reason I had to create buffer on my own and pass it to the channel, so I could reference it later and take directly from it instead of the channel, to achieve this flushing in my app.